### DISCUSSION TITLE
How can i add mutiple files into a github repository with one commit?

### DISCUSSION THREAD

**Discussion by tranphuocloc070699** [hub4j/github-api#1755] [Q&A]
As titled, i can implement code to add and commit into a github repository, but it create mutiple time push 
```
GHRepository repo = github.getRepository("tranphuocloc070699/new-repository");
 while (condition) {
                    repo.createContent()
                            .message("Adding file: " + filename)
                            .path(path)
                            .content(bytesContent)
                            .commit();
            }
```
Can anyone help me modify this code to add mutiple content into github repo with only one commit?

Thanks for all

**Comment by bitwiseman** [comment]
What one way to do this is with `GHTreeBuilder`:

https://github.com/hub4j/github-api/blob/998b6589823ee44111407725276584c20a21e54b/src/test/java/org/kohsuke/github/GHTreeBuilderTest.java#L119-L125

The call to `updateTree()` takes the creates the tree, attaches it to a commit, and then applies that commit to a branch:

https://github.com/hub4j/github-api/blob/998b6589823ee44111407725276584c20a21e54b/src/test/java/org/kohsuke/github/GHTreeBuilderTest.java#L178-L188

Hope that helps!
